### ISSUE TITLE
Performance issue:  drawCircles(context: CGContext) in LineChart

### ISSUE THREAD

**Issue by Osolemio** [danielgindi/Charts#3798]
Hello, Daniel
I used MPAndroidCharts with very good performance for my LineCharts (a number of lines up to 40 000 and more for each one). I very happy that you could migrate this library to iOS. Thank you very much.
I wonder why on iOS my charts rendering causes too long and heavy CPU load. So, rendering takes too much time comparing to MPAndroidCharts.

Under profiler I see that 100% eats drawCircles(), despite being disabled:

`lineDataSets[i].drawCirclesEnabled = false
  lineDataSets[i].drawCircleHoleEnabled = false //added that just for case`

![default](https://user-images.githubusercontent.com/1008281/50382438-05a1a080-06b0-11e9-90b8-909b37ca8dc0.png)

So, I remember very well that if I enable drawCircles for big datasets on Android it's getting slow as I have here with "disabled" circles.

P.S. 
` private func drawCircles(context: CGContext)
    {
        return`

saved me.

**Comment by liuxuan30** [comment]
I'm confused, 
>P.S.
private func drawCircles(context: CGContext) { return
saved me. All the heavy load (100%) now is on aa_render CoreGraphics. That's ok. The same chart rendering takes 2s. 20! times faster.

do you have issue now or solved?

**Comment by Osolemio** [comment]
> I'm confused,
> do you have issue now or solved?

The issue is still there. I have blocked non-disabled func drawCircles with 'return' in the beginning, because I don't need it in my package. The issue IMHO that it shouldn't been called when drawCirclesEnabled = false

**Comment by Osolemio** [comment]
> So we can skip some code, can you try it out if it solves your perf issue?

Yep, that's it. Can confirm. It fixes the issue. Moved it up in "if" statement.

Thank you! Have a nice day
